swe_roe/ex2b.c

```diff
@@ -0,0 +1,1754 @@
+    hul_vec_int[ii] = x_ptr[l * ndof + 1];
+    hvl_vec_int[ii] = x_ptr[l * ndof + 2];
+
+    hr_vec_int[ii]  = x_ptr[r * ndof + 0];
+    hur_vec_int[ii] = x_ptr[r * ndof + 1];
+    hvr_vec_int[ii] = x_ptr[r * ndof + 2];
+  }
+
+  // Compute u/v for left and right cells
+  PetscCall(GetVelocityFromMomentum(num, app->tiny_h, hl_vec_int, hul_vec_int, hvl_vec_int, ul_vec_int, vl_vec_int));
+  PetscCall(GetVelocityFromMomentum(num, app->tiny_h, hr_vec_int, hur_vec_int, hvr_vec_int, ur_vec_int, vr_vec_int));
+
+  // Update u/v for reflective internal edges
+  for (PetscInt ii = 0; ii < mesh->num_internal_edges; ii++) {
+    PetscInt  iedge      = edges->internal_edge_ids[ii];
+    PetscInt  cellOffset = edges->cell_offsets[iedge];
+    PetscInt  l          = edges->cell_ids[cellOffset];
+    PetscInt  r          = edges->cell_ids[cellOffset + 1];
+    PetscReal bl         = b_ptr[l];
+    PetscReal br         = b_ptr[r];
+
+    cn_vec_int[ii] = edges->cn[iedge];
+    sn_vec_int[ii] = edges->sn[iedge];
+
+    PetscBool is_edge_vertical;
+    if (PetscAbs(edges->normals[iedge].V[0]) < 1.e-10) {
+      is_edge_vertical = PETSC_TRUE;
+    } else if (PetscAbs(edges->normals[iedge].V[1]) < 1.e-10) {
+      is_edge_vertical = PETSC_FALSE;
+    } else {
+      printf("The code only support quad cells with edges that align with x and y axis\n");
+      exit(0);
+    }
+
+    if (bl == 1 && br == 0) {
+      // Update left values as it is a reflective boundary wall
+      hl_vec_int[ii] = hr_vec_int[ii];
+      if (is_edge_vertical) {
+        ul_vec_int[ii] = ur_vec_int[ii];
+        vl_vec_int[ii] = -vr_vec_int[ii];
+      } else {
+        ul_vec_int[ii] = -ur_vec_int[ii];
+        vl_vec_int[ii] = vr_vec_int[ii];
+      }
+
+    } else if (bl == 0 && br == 1) {
+      // Update right values as it is a reflective boundary wall
+      hr_vec_int[ii] = hl_vec_int[ii];
+      if (is_edge_vertical) {
+        ur_vec_int[ii] = ul_vec_int[ii];
+        vr_vec_int[ii] = -vl_vec_int[ii];
+      } else {
+        ur_vec_int[ii] = -ul_vec_int[ii];
+        vr_vec_int[ii] = vl_vec_int[ii];
+      }
+    }
+  }
+
+  // Call Riemann solver
+  PetscCall(solver(num, hl_vec_int, hr_vec_int, ul_vec_int, ur_vec_int, vl_vec_int, vr_vec_int, sn_vec_int, cn_vec_int, flux_vec_int, amax_vec_int));
+
+  // Save the flux values in the Vec based by TS
+  for (PetscInt ii = 0; ii < mesh->num_internal_edges; ii++) {
+    PetscInt  iedge      = edges->internal_edge_ids[ii];
+    PetscInt  cellOffset = edges->cell_offsets[iedge];
+    PetscInt  l          = edges->cell_ids[cellOffset];
+    PetscInt  r          = edges->cell_ids[cellOffset + 1];
+    PetscReal edgeLen    = edges->lengths[iedge];
+
+    PetscReal hl = x_ptr[l * ndof + 0];
+    PetscReal hr = x_ptr[r * ndof + 0];
+    PetscReal bl = b_ptr[l];
+    PetscReal br = b_ptr[r];
+
+    *amax_value = fmax(*amax_value, amax_vec_int[ii]);
+
+    if (bl == 0 && br == 0) {
+      // Both, left and right cells are not boundary walls
```

Comment:
Do you mean both left and right are not ghost cells?

Reply:
No. The code enters the loop if both left and right are not reflective boundary walls.

Reply:
Maybe we could use slightly more descriptive names for these, or leave a comment somewhere describing what they mean.